design/frontend-design.md

```diff
@@ -0,0 +1,178 @@
+## Test Cases
+1.  class with two fields @sitan, UI @yiming
+```
+    input:
+        class C:
+            a : int = 1
+            b : int = 2
+        c : C = None
+        c = C()
+    // in REPL:
+    >>> c
+    
+
+    output:
+        {
+            address: 100
+            a: 1
+            b: 2
+        }
+```
+
+2. class with nested class instance
+```
+   input:
+       class E(object):
+            a : int = 1
+       class C(object):
+            a : bool = True
+            e : E = None
+            def __init__(self: C):
+               self.e = E()
+            def d(self: C) -> int:
+              return 1
+        c : C = None
+        c = C()
+    // in REPL:
+    >>> c
+        
+    output:
+        {
+            address: 100
+            a: True
+            e: {
+                address: 108
+                a: 1
+            }
+        }
+```
+
+3.  class with inheritance: 
+ we need to wait for the inheritance group to finish their feature.
+```
+input: 
+    class E(object):
+        a : int = 1
+        f : int = 2
+    class C(E):
+        a : int = 2
+        e : E = None
+        def __init__(self: C):
+           self.e = E()
+        def d(self: C) -> int:
+          return 1
+    c : C = None
+    c = C()
+    // in REPL:
+    >>> c
+
+output: 
+      {
+            address: 100
+            a: 2,
+            f: 2,
+            e: {
+                address: 112
+                a: 1
+            }
+        }
+```
+
+4. nested instance with uninitilized instace
+```
+    input:
+    class E(object):
+        a : int = 1
+
+    class C(E):
+        a : int = 2
+        e : E = None
+        def d(self: C) -> int:
+            return 1
+    c : C = None
+    c = C()
+    // in REPL:
+    >>> c
+
+    output:
+        {
+            address: 0x1234
+            a: 1
+            e: none
+        }
+```
```

Comment:
good tests, but in general, I think you should test more than one operation in REPL, for example, do a couple of things in REPL and change some field values and then get an output.

Reply:
tested